plugins/net_plugin/net_plugin.cpp

```diff
@@ -1446,6 +1485,9 @@ namespace eosio {
    }
 
    void connection::enqueue_block( const signed_block_ptr& b, bool to_sync_queue) {
+      if(!is_participating()) {
```

Comment:
Need similar code (except with continue instead of return) also in dispatch_manager::bcast_block, after the if (!sb) block that sends the go away message, before block is sent to enque_buffer. Also in dispatch_manager::bcast_transaction, need to change:

if( cp->is_blocks_only_connection() || !cp->current() )

into:

if( cp->is_blocks_only_connection() || !cp->current() || !cp->is_participating() )

I believe this covers everything, but double check me that the "Need to check participating flag before calling enque_buffer for any transaction, block, or notice_messages for any connection." requirement has been met.

Reply:
In dispatch_manager::bcast_block, the "if(!sb)" block is in the lambda passed to for_each_block_connection.  So "return true" actually results in a continue to the next connection.

I did a search on all references to enqueue_buffer and each has a preceding call to is_participating() in the code to bypass the call to enqueue_buffer.